app/bundles/FormBundle/Model/FormModel.php

```diff
@@ -788,24 +788,31 @@ public function populateValuesWithGetParameters(Form $form, &$formHtml)
     public function populateValuesWithLead(Form $form, &$formHtml)
     {
         $formName = $form->generateFormName();
-        $lead     = $this->leadModel->getCurrentLead();
+        $fields   = $form->getFields();
+        /** @var \Mautic\FormBundle\Entity\Field $field */
+        foreach ($fields as $key=> $field) {
+            $leadField  = $field->getLeadField();
+            $isAutoFill = $field->getIsAutoFill();
+
+            if (!isset($leadField) || !$isAutoFill) {
+                unset($fields[$key]);
+            }
+        }
+        // no fields for populate
+        if (!count($fields)) {
+            return;
+        }
 
+        $lead     = $this->leadModel->getCurrentLead();
         if (!$lead instanceof Lead) {
             return;
         }
 
-        $fields = $form->getFields();
-        /** @var \Mautic\FormBundle\Entity\Field $f */
-        foreach ($fields as $f) {
-            $leadField  = $f->getLeadField();
-            $isAutoFill = $f->getIsAutoFill();
+        foreach ($fields as $field) {
+            $value = $lead->getFieldValue($leadField);
```

Comment:
This is still fetching the wrong field name. It's fetching the value of the field consumed in the foreach of 793. $leadField should be $field inside this loop.

Reply:
Fixed